### ISSUE TITLE
Archiving removes wrong files (when archive and log files are in the same directory)

### ISSUE THREAD

**Issue by StanislavSumyatin** [NLog/NLog#1670]
**Bug**
**NLog version**: 4.3.7
**Platform**: .Net 4.5.2

**Current NLog config** (xml or C#, if relevant)

``` xml
<target name="file" type="File"
                        layout="${longdate} | ${level:uppercase=true} }"
                        fileName="${LogFolder}/${LogSource}.log"
                        archiveFileName="${LogFolder}/${LogSource}-${date:format=yyyy-MM-dd}#{##}.log"
        archiveEvery="Hour"
                concurrentWrites="true"
        archiveNumbering="Rolling"
        maxArchiveFiles="84"/>

```

**Current result**: Not only "nlog files" are deleted from LogFolder, but also web.config and logs from other systems when method `DeleteOldArchiveFiles` is called

**Expected result**: Only nlog files are deleting when method DeleteOldArchiveFiles is called. 
Problem is that in method DynamicFileArchive.InitializeForArchiveFolderPath all files from directory LogFolder are queued and then deleting in method DeleteOldArchiveFiles.

Internal log:
2016-09-30 14:26:40.3940 Debug Fatal => file
2016-09-30 14:32:41.5089 Warn Cannot delete old archive file : 'C:\logs\SiteLogs\Test8\AvayaToolkit.log'. Exception: System.IO.IOException: The process cannot access the file 'C:\logs\SiteLogs\Test8\AvayaToolkit.log' because it is being used by another process.
   at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
   at System.IO.File.InternalDelete(String path, Boolean checkHost)
   at NLog.Targets.FileTarget.DynamicFileArchive.DeleteOldArchiveFiles()
2016-09-30 14:32:41.5089 Warn Cannot delete old archive file : 'C:\logs\SiteLogs\Test8\EpiServer_Test8.log'. Exception: System.IO.IOException: The process cannot access the file 'C:\logs\SiteLogs\Test8\EpiServer_Test8.log' because it is being used by another process.
   at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
   at System.IO.File.InternalDelete(String path, Boolean checkHost)
   at NLog.Targets.FileTarget.DynamicFileArchive.DeleteOldArchiveFiles()

It deletes files AvayaToolkit.log and web.config from diretory.

**Comment by 304NotModified** [comment]
Could you give examples of filenames which are deleted?

Could you check and post the internal log? (level trace)

**Comment by 304NotModified** [comment]
probably the same

https://github.com/NLog/NLog/issues/1681

**Comment by 304NotModified** [comment]
tested this with #1681. There is indeed no mask currently on the archive files. 

Creating this mask is easy: archiveFileName="c:/temp/logsource1-2016-10-10-**.log", but this isn't  archiveFileName="c:/temp/logsource1-****-**-**-**.log", but this isn't.

I did not tested this case in-depth yet, but could you test with a filename without `#`? 

So `${LogFolder}/${LogSource}-${date:format=yyyy-MM-dd}#{##}.log` => `${LogFolder}/${LogSource}-${date:format=yyyy-MM-dd}_{##}.log`

**Comment by 304NotModified** [comment]
This is an issue, but it's not trivial to fix.

**Comment by alexiuss** [comment]
Some testing results:
1. wrong files are deleting without # , without {##}
2. are deleting when archived files are in the dedicated directory and there are more than one target-rule
3. When “archiveFileName” like ${LogSource}-${date:format=yyyy-MM-dd}_{##} .log and new day comes, nlog starts rolling files with new day index and leaves with previous.
4. No logging of wrong files deleting even in internalLogLevel=Trace (one time we captured was when outlier file was locked “Warn Cannot delete old archive file”)

So for now we put maxArchiveFiles="0" to avoid deleting.

**Comment by 304NotModified** [comment]
Thanks for the testing results! 

Funny as it seems this never could have worked (when checking the git history)

**Comment by snakefoot** [comment]
Very strange that DynamicFileArchive.InitializeForArchiveFolderPath is activated, when having '{#}' as part of the archiveFileName.

Would this configuration  not work:

```
<target name="file" type="File"
         layout="${longdate} | ${level:uppercase=true} }"
         fileName="${LogFolder}/${LogSource}.log"
         archiveFileName="${LogFolder}/${LogSource}-{#}.log"
         archiveEvery="Hour"
         concurrentWrites="true"
         archiveDateFormat="yyyy-MM-dd"
         archiveNumbering="DateAndSequence"
         maxArchiveFiles="84"/>
```

Made the following changes:

- archiveNumbering="DateAndSequence"
- archiveDateFormat="yyyy-MM-dd"
- archiveFileName="${LogFolder}/${LogSource}-{#}.log"

When using archiveNumbering="Rolling" (or Sequence), then it expects the archiveFileName to be completely static (Thus using date-renderer is not supported).

**Comment by 304NotModified** [comment]
fixed by #1993, NLog 4.5. Thanks @snakefoot !
